### DISCUSSION TITLE
pymort specs

### DISCUSSION THREAD

**Discussion by fumitoh** [actuarialopensource/pymort#6] [Ideas]
Great work!
Here're some points that I think may be worth considering.

- It'd be good to have another way to specify `tableName` other than passing a full table name especially for those who don't use VS code.
- Having an option for `load` to return a table as a DataFrame with MutiIndex would be nice.
- I wonder how often the SOA's site is updated. If it's often, then I'd place the metadata somewhere on the web (like a gihub repo) and allow this tool to pull the update metadata. 
- This isn't easy, but having this tool work with Python 3.6 and 3.7 would gain more users.

**Comment by MatthewCaseres** [comment]
Thanks, I didn't get a notification for you starting this discussion so it took me a minute to see this.

1. You can actually just pass in any string that ends with the proper table number. So `"1"` is valid. I strip numbers from the end of the string using a regex.
2. This sort of relates to the need to "longify" the mortality tables, so that they are proper tables.
3. It isn't hard to download all the files again when needed. Can be automated but not super sure it is worthwhile.
4. The literal types I am using to power autocompletion are from 3.8. I use 3.9 in the pyproject.toml but this is probably not right, I believe ^3.8 will suffice. I will probably keep it at 3.8, options like pyenv make this not too terrible a thing to put upon the user.

The biggest issue for me is that many tables are irregular and the current API is broken for them. The second biggest issue is providing normalized tables and collections of tables for whole table sets.

Thanks for the star. Seeing a couple of people take an interest makes me want to ship the changes that are necessary. I will try to close out all the open issues, hopefully something that looks like 1.0 is available by the end of October.

**Reply by MatthewCaseres** [reply]
This idea about the MultiIndex is very useful. I was wondering how to represent the tables and I was thinking something ridiculous like 

```py
Union[dict[int, float], dict[int, dict[int, float]]]
```

The MultiIndex is a big improvement over this. Thanks.
